Makefile

```diff
@@ -75,71 +75,86 @@ HOST_BACKEND = 1
   CFLAGS += -DSECVAR_GUEST_BACKEND
 endif
 
 _LDFLAGS += -L./lib
 
-SECVARCTL_SRCS = ./src/generic.c \
-                 ./src/secvarctl.c
+SRCS = generic.c \
+       secvarctl.c
+
+# TODO: Consider splitting this also into its own Makefile.inc?
+EXTERNAL_SRCS = external/extraMbedtls/pkcs7.c                                \
+                external/extraMbedtls/pkcs7_write.c                          \
+                external/skiboot/libstb/secvar/secvar_util.c                 \
+                external/skiboot/libstb/secvar/crypto/crypto-mbedtls.c       \
+                external/skiboot/libstb/secvar/crypto/crypto-openssl.c       \
+                external/skiboot/libstb/secvar/crypto/crypto-gnutls.c        \
+                external/skiboot/libstb/secvar/backend/edk2-compat.c         \
+                external/skiboot/libstb/secvar/backend/edk2-compat-process.c
+
+include backends/host/Makefile.inc
+include backends/guest/Makefile.inc
+
+SRCS += $(EXTERNAL_SRCS)
 
-SECVARCTL_OBJS = $(SECVARCTL_SRCS:.c=.o)
-OBJCOV = $(patsubst %.o, %.cov.o,$(SECVARCTL_OBJS))
+SRCS += $(addprefix backends/host/,$(HOST_SRCS))
+SRCS += $(addprefix backends/guest/,$(GUEST_SRCS))
+
+OBJS = $(addprefix $(OBJ_DIR)/,$(SRCS:.c=.o))
+OBJCOV = $(patsubst %.o, %.cov.o,$(OBJS))
 
 _CFLAGS += $(CFLAGS) $(INCLUDES)
 _LDFLAGS += $(LDFLAGS)
 
 export CFLAGS
 export LDFLAGS
 
-all: secvarctl-backends secvarctl secvarctl-cov
+all: secvarctl
 
-coverage: secvarctl-backends secvarctl-cov
+coverage: secvarctl-cov
 
-secvarctl: $(SECVARCTL_OBJS)
-	$(CC) $(_CFLAGS) $(STATICFLAG) $^ -o $(BIN_DIR)/$@ $(_LDFLAGS)
-	@echo "secvarctl Build successful!"
+$(OBJ_DIR)/%.o: %.c
+	@mkdir -p $(dir $@)
+	$(CC) $(_CFLAGS) $< -o $@ -c
 
-secvarctl-cov: $(OBJCOV)
-	$(CC) $(_CFLAGS) $^ $(STATICFLAG) -fprofile-arcs -ftest-coverage -o $(BIN_DIR)/$@ $(_LDFLAGS)
-	@echo "secvarctl-cov Build successful!"
+$(OBJ_DIR)/%.cov.o: %.c
+	@mkdir -p $(dir $@)
+	$(CC) $(_CFLAGS) -c --coverage $< -o $@
 
-secvarctl-backends:
-	@mkdir -p $(LIB_DIR)
-ifeq ($(strip $(HOST_BACKEND)), 1)
-	@$(MAKE) -C $(HOST_BACKEND_DIR) LIB_DIR=$(LIB_DIR) OPENSSL=$(OPENSSL) GNUTLS=$(GNUTLS) \
-	            MBEDTLS=$(MBEDTLS) CRYPTO_READ_ONLY=$(CRYPTO_READ_ONLY) DEBUG=$(DEBUG) \
-	            DYNAMIC_LIB=$(DYNAMIC_LIB)
-endif
-ifeq ($(strip $(GUEST_BACKEND)), 1)
-	@$(MAKE) -C $(GUEST_BACKEND_DIR) DEBUG=$(DEBUG) LIB_DIR=$(LIB_DIR) \
-	            CRYPTO_READ_ONLY=$(CRYPTO_READ_ONLY) DYNAMIC_LIB=$(DYNAMIC_LIB)
-endif
+$(BIN_DIR)/secvarctl: $(OBJS) $(LIBSTB_SECVAR)
+	@mkdir -p $(BIN_DIR)
+	$(CC) $(_CFLAGS) $(STATICFLAG) $^ -o $@ $(_LDFLAGS) -lcrypto -lmbedtls -lmbedx509 -lmbedcrypto
 
-%.o: %.c
-	$(CC) $(_CFLAGS) $< -o $@ -c
+$(BIN_DIR)/secvarctl-cov: $(OBJCOV) $(LIBSTB_SECVAR)
+	@mkdir -p $(BIN_DIR)
+	$(CC) $(_CFLAGS) $^ $(STATICFLAG) -fprofile-arcs -ftest-coverage -o $@ $(_LDFLAGS)
+
+$(LIBSTB_SECVAR):
+	$(MAKE) CFLAGS=-DSECVAR_CRYPTO_WRITE_FUNC -C external/libstb-secvar lib/libstb-secvar-openssl.a
```

Comment:
Right now, only OpenSSL works because for some reason that's the only crypto backend in libstb-secvar. Also, I don't see a point in having `SECVAR_CRYPTO_WRITE_FUNC` and `CRYPTO_READ_ONLY`. Either we should consolidate them into one flag, or remove it entirely and simplify the build. I'm not a fan of having so many build-time variables, it creates way too many possibly build combinations.